### ISSUE TITLE
App crashes when requesting biometric verification while the phone is rotated

### ISSUE THREAD

**Issue by MarvashMagalli** [tachiyomiorg/tachiyomi#8896]
### Steps to reproduce

1. Open the app
2. Wait for the "touch the fingerprint sensor" message when the "require unlock" option is enabled
3. Rotate the phone
4. App crashes

App also crashes if you open the app while the phone is already rotated and the same settings are active.

https://user-images.githubusercontent.com/58483315/212030716-a0931477-e9ca-4861-91d8-f66e5fa830a5.mp4

### Expected behavior

The app shouldn't crash

### Actual behavior

The app crashes instantly

### Crash logs

[tachiyomi_crash_logs.txt](https://github.com/tachiyomiorg/tachiyomi/files/10400125/tachiyomi_crash_logs.txt)

### Tachiyomi version

0.14.3

### Android version

Android 8.0.0

### Device

SM-G935F (Samsung Galaxy S7 Edge)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.14.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by Dari8987898** [comment]
I did a bit of research, but before starting:

- Note 1 - i did this research alone and i'm not very familiar with android developement so i'll make sure to be as detaild as possible (hence, wall of text incoming);
- Note 2 - this issue is raleted to [app/src/main/java/eu/kanade/tachiyomi/ui/security/UnlockActivity.kt](https://github.com/tachiyomiorg/tachiyomi/blob/master/app/src/main/java/eu/kanade/tachiyomi/ui/security/UnlockActivity.kt), in particular to the `androidx` package.

In the official documentation of the `androidx` package it's specified that "[If you want to use `androidx`-namespaced libraries in a new project, you need to set the compile SDK to Android 9.0 (API level 28) or higher](https://developer.android.com/jetpack/androidx#using_androidx_libraries_in_your_project)", and that should be the reason why your device (which is using android 8) is getting the error.

I also found that "[`androidx` artifacts with version 1.0.0 are binary equivalent to the Support Library 28.0.0 artifacts](https://developer.android.com/jetpack/androidx/migrate#prerequisites)". Due to this, the android versions prior to the 9.0 don't get access to the packages/features of `androidx` (that are required to make the fingerprint unlock work). Also [the library 28.0.0 is the last release under the `android.support` packaging](https://developer.android.com/topic/libraries/support-library/revisions#28-0-0) so there is no way that this feature will be backwords-compatible, because it's an `androidx` specific feature.

I hope that this will be of some help.

**Comment by Dari8987898** [comment]
I also found that this should be related to #5695

**Comment by MarvashMagalli** [comment]
Adding a note, the fingerprint unlock works fine if I don't rotate the phone, always did. I rarely rotate the phone, so I don't know if this would have been a problem in previous app versions or not.

**Comment by stevenyomi** [comment]
@Dari8987898 No, this shouldn't an issue with the `androidx` package. The compile SDK is just for compiling the app and doesn't affect running on older versions (in fact the app is currently compiled with A13 SDK). Also the `android.support` package is just the old version of `androidx` - it's just Google decided to move the system-unrelated packages (some of which help provide compatibility with older Android versions) to another namespace (`androidx`).

**Comment by stevenyomi** [comment]
Upstream issue: https://issuetracker.google.com/issues/181805603

**Comment by Dari8987898** [comment]
@stevenyomi thanks for the explanation, i misunderstood what the android manual said. It makes sense, it's just that the 'renaming' of the package seemed strange to me

**Comment by MarvashMagalli** [comment]
My S7 broke, I am currently using an S6, so I tested this again and here's my findings:

- If my phone is already rotated, then I can access the app just fine using my fingerprint.
- If I rotate the phone the moment I'm being asked to input my fingerprint, the app closes, with no report.
- If my phone is already rotated, and I open the app, and while on the fingerprint request screen I turn the phone right side up again, the app closes, with no report.

The app version is again 0.14.3

I will update my experience once again after my new android phone arrives.

**Comment by MarvashMagalli** [comment]
Tested on a Samsung S22 Ultra, Android 13, and the problem is not present.
